api/samples/CMakeLists.txt

```diff
@@ -204,54 +211,54 @@ configure_DynamoRIO_global(OFF ON)
 
 # Use ;-separated lists for source files and extensions.
 
-add_sample_client(bbbuf       "bbbuf.c"         "drmgr;drreg;drx")
-add_sample_client(bbcount     "bbcount.c"       "drmgr;drreg;drx")
-add_sample_client(bbsize      "bbsize.c"        "drmgr;drx")
-add_sample_client(div         "div.c"           "drmgr")
-add_sample_client(empty       "empty.c"         "")
-add_sample_client(memtrace_simple "memtrace_simple.c;utils.c" "drmgr;drreg;drutil;drx")
-add_sample_client(memval_simple   "memval_simple.c;utils.c"   "drmgr;drreg;drutil;drx")
-add_sample_client(instrace_simple "instrace_simple.c;utils.c" "drmgr;drreg;drx")
-add_sample_client(opcode_count    "opcode_count.cpp"    "drmgr;drreg;drx;droption")
+add_sample_client(bbbuf       "bbbuf.c"         "drmgr;drreg;drx" OFF)
+add_sample_client(bbcount     "bbcount.c"       "drmgr;drreg;drx" OFF)
+add_sample_client(bbsize      "bbsize.c"        "drmgr;drx" OFF)
+add_sample_client(div         "div.c"           "drmgr" OFF)
+add_sample_client(empty       "empty.c"         "" OFF)
+add_sample_client(memtrace_simple "memtrace_simple.c;utils.c" "drmgr;drreg;drutil;drx" OFF)
+add_sample_client(memval_simple   "memval_simple.c;utils.c"   "drmgr;drreg;drutil;drx" OFF)
+add_sample_client(instrace_simple "instrace_simple.c;utils.c" "drmgr;drreg;drx" OFF)
+add_sample_client(opcode_count    "opcode_count.cpp"    "drmgr;drreg;drx;droption" ON)
```

Comment:
Hmm, multiple thoughts here:

1. This ignores the user's SHOW_RESULTS option which seems undesirable -- although that option should maybe rarely be used and nobody should rely on it (should we mark it ADVANCED?) (except Windows where annoying popups could make it a reasonable user option -- see next tiem).  I guess I'm just complaining and not suggesting a solution: label SHOW_RESULTS and not affecting all samples?  And mark it ADVANCED and make it not really part of the user-facing choices or sthg.

2. This will cause popups on Windows which can hang automated tests.  Maybe long-term what we want is to get rid of SHOW_OPTIONS, have all samples print to the console, add functionality-testing tests for all samples, and have a USE_POPUPS or sthg option for Windows where console printing doesn't work for GUI apps and pre-Win8 where there are all kinds of console printing problems.

Reply:
Yes, it does ignore the SHOW_RESULTS option when building tests. This seems reasonable no? Otherwise we cannot set the up, or do you mean if SHOW_RESULTS is false, we switch back to just sanity tests?

Reply:
I guess I'm seeing downsides in:
+ The extra add_sample_client param, which is made public to the user, right?  This is our sample CMakeLists.txt for users.
+ Having SHOW_RESULTS overridden in a potentially confusing way: just some samples, and then just UNIX, and not lining up with which samples have tests: this test is 32-bit only, but this sample will print results in 64-bit.

Here is my proposal:
+ Create two build executable targets for each sample: one public-facing with SHOW_RESULTS on, and another that's for testing purposes.  Then we also avoid samples being silent in a build dir with BUILD_TESTS on.  As for tests printing or not: I guess as we add functional tests and remove sanity tests (is that what we decided?) we flip them to print vs not print.  Long-term the public ones and testing ones are identical on UNIX probably, and on Windows it's popup vs console printing.

Reply:
Okay, I agree, will make the code simpler with fewer branches. So in this case, we need to ensure that the second build is indeed not exported to release packages.

Reply:
>  Long-term the public ones and testing ones are identical on UNIX probably

Do you mean eventually we will move away from the SHOW_RESULTS option and always print the results?

Reply:
> Do you mean eventually we will move away from the SHOW_RESULTS option and always print the results?

If we have a separate build target for tests, we could get rid of the SHOW_RESULTS CMake option, yes.  The only reason I could see to have an option would be for the Windows popups which can be annoying: but these are just samples and people are supposed to edit them so maybe no reason to go adding runtime or build options for that.

Reply:
Done.